### PR TITLE
[Python] Port Number Recognizer to Python

### PR DESCRIPTION
**Pull request by amitstein**
**This PR includes:**
- Number Recognizer (Number, Ordinal and Percent models) for languages: English, Chinese, Spanish, Portuguese and French
- The resource generation tool
- Data-driven test engine
- Sample: simple-console

### CODE REVIEW

File: .NET/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Getting Started
 
-Recognizer's are organized into groups and designed to be used in C# and Node.js to help you build great applications! To use the samples clone our GitHub repository using Git.
+Recognizer's are organized into groups and designed to be used in C#, Node.js and Python to help you build great applications! To use the samples clone our GitHub repository using Git.
```
Comment (tellarin) [inline .NET/README.md]: Nit: I'm always partial to the Oxford comma (C#, Node.js, and Python) ;-)

File: Python/samples/simple_console/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+../../libraries/recognizers-number/
```
Comment (tellarin) [inline Python/samples/simple_console/requirements.txt]: Is this really just a txt file? Just curious.

File: README.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ Microsoft.Recognizers.Text powers pre-built entities in both [**LUIS: Language U
 The Microsoft.Recognizers.Text packages currently target two platforms:
 * [C# version](https://github.com/Microsoft/Recognizers-Text/tree/master/.NET) - **NuGet packages** available at: https://www.nuget.org/profiles/Recognizers.Text
 * [JavaScript verison](https://github.com/Microsoft/Recognizers-Text/tree/master/JavaScript/packages/recognizers-text-suite) - **NPM packages** available at: https://www.npmjs.com/~recognizers.text
+* [Python version](https://github.com/Microsoft/Recognizers-Text/tree/master/Python)
```
Comment (tellarin) [inline README.md]: Are you familiar with the process to release PyPI packages? I'm not. :)

File: Specs/DateTime/German/DateTimePeriodParser.json [hunk 4]
```diff
@@ -48,7 +48,7 @@
       }
     ]
   },
-  {//Parser returns NULL as value. The Parser Returns Null as value. Should not be different to the one above
```
Comment (tellarin) [inline Specs/DateTime/German/DateTimePeriodParser.json]: Let's not remove these annotations unless they really no longer apply.
If they are still valid, we can move them into a json attribute called "TestComment".

File: Specs/Number/Chinese/NumberModel.json [hunk 5]
```diff
@@ -460,8 +427,7 @@
 ,
 {
   "Input": "负10000",
-  "NotSupportedByDesign": "python",
-  "NotSupported": "javascript",
+  "NotSupported": "javascript, python",
```
Comment (tellarin) [inline Specs/Number/Chinese/NumberModel.json]: Any reason this specific case is problematic? I forget if we've discussed it for Javascript.

File: Specs/Number/Chinese/NumberModel.json [hunk 6]
```diff
@@ -3269,7 +3048,7 @@
 ,
 {
   "Input": "三 百 五 又 三 分之 一",
-  "NotSupportedByDesign": "python",
+  "Debug": true,
```
Comment (tellarin) [inline Specs/Number/Chinese/NumberModel.json]: Please remove the Debug tags.

File: Specs/Number/Chinese/PercentModel.json [hunk 7]
```diff
@@ -1610,7 +1498,7 @@
 ,
 {
   "Input": "四點 五個百分點",
-  "NotSupportedByDesign": "python",
+  "Debug": true,
```
Comment (tellarin) [inline Specs/Number/Chinese/PercentModel.json]: Same

File: Specs/NumberWithUnit/German/CurrencyModel.json [hunk 8]
```diff
@@ -286,7 +286,7 @@
   ]
 }
 ,
-{//Age-Model currently does not work with ordinals
```
Comment (tellarin) [inline Specs/NumberWithUnit/German/CurrencyModel.json]: Same comment as in the datetime specs.

File: Specs/Number/Chinese/NumberModel.json (hunk 5 above)
Comment (enzocanoo) [reply Specs/Number/Chinese/NumberModel.json]: Fixed! (based on #362)

File: Specs/NumberWithUnit/German/CurrencyModel.json (hunk 8 above)
Comment (enzocanoo) [reply Specs/NumberWithUnit/German/CurrencyModel.json]: Fixed!
Python does not support JSON files with c-style comments so we added a new JSON property named **Comment**

File: Python/samples/simple_console/requirements.txt (hunk 2 above)
Comment (enzocanoo) [reply Python/samples/simple_console/requirements.txt]: Yes, this file contains a list of items to be installed using `pip install -r requirements.txt`. [(more information)](https://pip.pypa.io/en/stable/user_guide/#requirements-files)

**Pull request merged**